spring-cloud-task-docs/src/main/asciidoc/features.adoc

```diff
@@ -0,0 +1,170 @@
+|The name for the task as determined by the configured `TaskNameResolver`.
+
+|`starTime`
+|The time the task was started as indicated by the `ContextRefreshEvent`.
+
+|`endTime`
+|The time the task was completed as indicated by the `ContextClosedEvent`.
+
+|`exitMessage`
+|Any information available at the time of exit.  If an exception is the cause of the end
+of the task (as indicated via an `ApplicationFailedEvent`), the stack trace for that
+exception will be stored here.
+
+|`parameters`
+|A `List` of the string parameters as they were passed into the executable boot
+application.
+|===
+
+[[features-lifecycle-exit-codes]]
+=== Mapping Exit Codes
+
+When a task completes, it will want to return an exit code to the OS.  If we take a look
+at our original example, we can see that we are not controlling that aspect of our
+application. So if an exception is thrown, the JVM will return a code that may or may not
+be of any use to you in the debugging of that.
+
+As such, Spring Boot provides an interface, `ExitCodeExceptionMapper` that allows you to
+map uncaught exceptions to exit codes.  This allows you to be able to indicate at that
+level what went wrong.  Also, by mapping exit codes in this manor, Spring Cloud Task will
+record the exit code returned.
+
+[[features-configuration]]
+== Configuration
+
+Spring Cloud Task provides an out of the box configuration as defined in the
+ `DefaultTaskConfigurer` and `SimpleTaskConfiguration`.  This section will walk through
+the defaults as well as how to customize Spring Cloud Task for your needs
+
+[[features-data-source]]
+=== DataSource
+
+Spring Cloud Task utilizes a datasource for storing the results of task executions.  By
+default, we provide an in memory instance of H2 to provide a simple method of
+bootstrapping development.  However, in a production environment, you'll want to configure
+your own `DataSource`.
+
+If your application utilizes only a single `DataSource` and that will serve as both your
+business schema as well as the task repository, all you need to do is provide any
+`DataSource` (via Spring Boot's configuration conventions is the easiest way).  This will
+be automatically used by Spring Cloud Task for the repository.
+
+If your application utilizes more than one `DataSource`, you'll need to configure the
+task repository with the appropriate `DataSource`.  This customization can be done via an
+implementation of the `TaskConfigurer`.
+
+[[features-task-configurer]]
+=== TaskConfigurer
+
+The `TaskConfigurer` is a strategy interface allowing for users to customize the way
+components of Spring Cloud Task are configured.  By default, we provide the
+`DefaultTaskConfigurer` that provides logical defaults (`Map` based in memory components
+useful for development if no `DataSource` is provided and JDBC based components if there
+is a `DataSource` available.
+
+The `TaskConfigurer` allows the configuration of three main components:
+
+|===
+|Component |Description |Default
```

Comment:
Can we add this to the `Default` header.  `Default (provided by DefaultTaskConfigurer)` ?